File: matlab/example_mHT_optional_parameters.m
% Add subfolders to PATH variable (can be deleted if package is already in
% folder within PATH or files are in the same catalog as this script).
addpath(fullfile(fileparts(which(mfilename)),"mHT"));

% Specifing display format output to double precision
format long;

% Example parameters of the S(1) 3-0 line of H2 perturbed by He.
% Reference: 10.1103/PhysRevA.101.052705.
nu0     = 112265.5949; % Unperturbed line position in cm-1.
GammaD  =     35.1e-3; % Doppler broadening in cm-1.
Gamma0  =     11.7e-3; % Speed-averaged line-width in cm-1.
Gamma2  =      5.4e-3; % Quadratic speed dependence parameter of the line-width in cm-1.
Delta0  =     30.5e-3; % Speed-averaged line-shift in cm-1.
Delta2  =     12.4e-3; % Quadratic speed dependence parameter of the line-shift in cm-1.
NuOptRe =     38.0e-3; % Real part of the Dicke parameter in cm-1.
NuOptIm =    -17.5e-3; % Imaginary part of the Dicke parameter in cm-1.

% Example frequency point:
nu = nu0 + 1; % Current wavenumber of the computation in cm-1.

% Optional parameters
Ylm   = 1.0e-3; % Dimensionless 
Xlm   = 0.5e-3; % Dimensionless
alpha = 2 ;     % Dimensionless

% Displaying mHT function output with no optional parameters
[outRe, outIm]  = profile(nu0, GamD, Gamma0, Gamma2, Delta0, Delta2, NuOptRe, NuOptIm, nu);
disp('The output of the mHT function with optional parameters:');
disp([outRe, outIm]);

% Displaying mHT function output with Ylm optional parameter only
[outRe, outIm]  = profile(nu0, GamD, Gamma0, Gamma2, Delta0, Delta2, NuOptRe, NuOptIm, nu, Ylm);
disp('The output of the mHT function with optional parameters (Ylm):');
disp([outRe, outIm]);

% Displaying mHT function output with alpha optional parameter only
[outRe, outIm]  = profile(nu0, GamD, Gamma0, Gamma2, Delta0, Delta2, NuOptRe, NuOptIm, nu, 0.0, 0.0, alpha);
disp('The output of the mHT function with optional parameters (alpha):');
disp([outRe, outIm]);

% Displaying mHT function output with optional parameters
[outRe, outIm]  = profile(nu0, GamD, Gamma0, Gamma2, Delta0, Delta2, NuOptRe, NuOptIm, nu, Ylm, Xlm, alpha);
disp('The output of the mHT function with optional parameters (all):');
disp([outRe, outIm]);

File: fortran77/example_mHT_optional_parameters.f
      program example_mHT_optional_parameters
c Example parameters of the S(1) 3-0 line of H2 perturbed by He (reference: 10.1103/PhysRevA.101.052705)
      implicit none
      double precision :: absorption,dispersion, mHTprofile
      double precision :: nu0 = 112265.5949d0 ! Unperturbed line position in cm-1.
      double precision :: GammaD = 35.1d-3 ! Doppler broadening in cm-1.
      double precision :: Gamma0 = 11.7d-3 ! Speed-averaged line-width in cm-1.
      double precision :: Gamma2 = 5.4d-3 ! Speed dependence of the line-width in cm-1.
      double precision :: Delta0 = 30.5d-3 ! Speed-averaged line-shift in cm-1.
      double precision :: Delta2 = 12.4d-3 ! Speed dependence of the line-shift in cm-1.
      double precision :: NuOptRe = 38.0d-3 ! Real part of the Dicke parameter in cm-1.
      double precision :: NuOptIm = -17.5d-3 ! Imaginary part of the Dicke parameter in cm-1.
      double precision :: nu ! Current wavenumber of the computation in cm-1.
      double precision :: Ylm = 0d0 ! Imaginary part of the 1st order (Rosenkranz) line mixing coefficients
      double precision :: Xlm = 0d0 ! Real part of the 1st order (Rosenkranz) line mixing coefficients
      double precision :: alpha = 10d0 ! perturber-to-absorber mass ratio
      logical :: calculate_dispersion=.false. ! Only the absorption profile will be calculated
   
      nu  = nu0 + 1d0
      absorption = mHTprofile(nu0,GammaD,Gamma0,Gamma2,Delta0,Delta2,
     &            NuOptRe,NuOptIm,nu,Ylm,Xlm,alpha,calculate_dispersion)
      calculate_dispersion = .true.
      dispersion = mHTprofile(nu0,GammaD,Gamma0,Gamma2,Delta0,Delta2,
     &            NuOptRe,NuOptIm,nu,Ylm,Xlm,alpha,calculate_dispersion)
      write(*, 101) absorption, dispersion
 101  format('The output of the mHT function (He-perturbed S(1)',
     &       ' 3-0 line in H2):',25X, 2F22.15)
c optional parameters
      Ylm = 1.0d-3
      Xlm = 0.5d-3
      alpha = 2.0d0
      calculate_dispersion = .false.
      
      absorption = mHTprofile(nu0,GammaD,Gamma0,Gamma2,Delta0,Delta2,
     &            NuOptRe,NuOptIm,nu,Ylm,Xlm,alpha,calculate_dispersion)
     
      calculate_dispersion = .true.
      dispersion = mHTprofile(nu0,GammaD,Gamma0,Gamma2,Delta0,Delta2,
     &            NuOptRe,NuOptIm,nu,Ylm,Xlm,alpha,calculate_dispersion)

      write(*, 102) absorption, dispersion
 102  format('The output of the mHT function with optional parameters ',
     &       '(He-perturbed S(1) 3-0 line in H2):', 2F22.15)

      end program example_mHT_optional_parameters
